soroban-env-host/src/budget.rs

```diff
@@ -445,6 +449,29 @@ impl AsBudget for &Host {
     }
 }
 
+impl DepthLimiter for Budget {
+    type DepthError = HostError;
```

Comment:
What you said makes sense. But it is a little inconvenient to work with non-HostError type because.
1. The `enter()` could result in `try_borrow_mut` error which is a different code. So to enrich the error on the host side we need to match the error code with the right one. 
2. All other budget functions (e.g. comparison, charge) already work with HostError. And we need to enclose a call `f` inside `with_limited_depth(f: FnOnce(&mut Self) -> Result(_, HostError)) -> Result<_, Error>`. which will require unnecessary error conversions.

Reply:
fair enough